### ISSUE TITLE
False positive on function's __signature__ attribute (reportFunctionMemberAccess)

### ISSUE THREAD

**Issue by leogermond** [microsoft/pyright#10413]
**Describe the bug**

[PEP-362](https://peps.python.org/pep-0362/) specifies that in order to cache the signature of any function, it is possible for the user to set its `__signature__` attribute.

When setting it, however, Pyright complains that the attribute isn't assignable.

I'm not sure if this belongs to typeshed or here, sorry if this ticket is misplaced.

**Code or Screenshots**

```python
❯ cat main.py
from inspect import signature


def main():

    def f():
        pass

    f.__signature__ = signature(f)


if __name__ == "__main__":
    main()
```

```
❯ ur pyright
/Users/leogermond/Lab/python/signature/main.py
  /Users/leogermond/Lab/python/signature/main.py:9:7 - error: Cannot assign to attribute "__signature__" for class "function"
    Attribute "__signature__" is unknown (reportFunctionMemberAccess)
1 error, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
Pyright uses the class definition for `builtins.function` defined in typeshed [here](https://github.com/python/typeshed/blob/a4d3f6f0de94d0beb98cb478eb031aecf5badce9/stdlib/builtins.pyi#L1009). It doesn't have an attribute named `__signature__`. If you think that it should include such a signature, please file an issue or a PR in the typeshed project.

**Comment by leogermond** [comment]
Thanks for the directions ! I had a hard time navigating the typeshed codebase. I opened an issue over there. Cheers ! :)
